shared/react-native/ios/KeybaseShare/FilesViewController.m

```diff
@@ -0,0 +1,222 @@
+//
+//  FilesViewController.m
+//  KeybaseShare
+//
+//  Created by John Zila on 10/25/18.
+//  Copyright © 2018 Keybase. All rights reserved.
+//
+
+#import "FilesViewController.h"
+#import "keybase/keybase.h"
+
+@interface FilesViewController ()
+@property (nonatomic) NSString* path; // the path we are currently showing
+@property NSArray* directoryEntries; // the directory entries at the current path
+@end
+
+NSString* const UpOneLevel = @"⤴ [up one level]";
+
+@implementation FilesViewController
+
+@synthesize path = _path;
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  
+  self.preferredContentSize = CGSizeMake(self.view.frame.size.width, 2*self.view.frame.size.height); // expand
+  self.definesPresentationContext = YES;
+  [self setPath:@"/"];
+  [self dispatchFilesBrowser];
+}
+
+- (void)dispatchFilesBrowser {
+  // show this spinner on top of the table view until we have processed the files.
+  UIActivityIndicatorView* av = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+  [self.view addSubview:av];
+  [av setTranslatesAutoresizingMaskIntoConstraints:NO];
+  [av setHidesWhenStopped:YES];
+  [av bringSubviewToFront:self.view];
+  [av startAnimating];
+  [self.tableView addConstraints:@[
+                                   [NSLayoutConstraint constraintWithItem:av
+                                                                attribute:NSLayoutAttributeCenterX
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem:self.tableView
+                                                                attribute:NSLayoutAttributeCenterX
+                                                               multiplier:1 constant:0],
+                                   [NSLayoutConstraint constraintWithItem:av
+                                                                attribute:NSLayoutAttributeCenterY
+                                                                relatedBy:NSLayoutRelationEqual
+                                                                   toItem:self.tableView
+                                                                attribute:NSLayoutAttributeCenterY
+                                                               multiplier:1 constant:0]
+                                   ]
+   ];
+  
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    NSError* error = NULL;
+    [self setDirectoryEntries:[NSArray new]];
+    NSString* jsonFiles = KeybaseExtensionListPath(self.path, &error); // returns the path list in JSON format
+    if (jsonFiles == nil) {
+      dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
you maybe just remove this and skip parseFiles if its null and just do the following dispatch_async (the extra reloadData will do nothing right)